### DISCUSSION TITLE
termux support (AnLinux, Andronix, {proot} arm64)

### DISCUSSION THREAD

**Discussion by SayaAEP** [ptitSeb/box64#238] [Q&A]
is it possible to run box64 in **termux** proot?
I think it seems **possible** because I saw this tutorial https://www.youtube.com/watch?v=TMV3Fp5YhTs

**Comment by SayaAEP** [comment]
proot is a virtual machine run by termux (or any other support application : AnLinux, Andronix {available on playstore}...)

**Comment by SayaAEP** [comment]
> is it possible to run box64 in **termux** proot? I think it seems **possible** because I saw this tutorial https://www.youtube.com/watch?v=TMV3Fp5YhTs

![vmter,mux](https://user-images.githubusercontent.com/91467886/154807670-23f0e77b-a42d-45b3-b79b-2a4dd90f24a6.PNG)

**Comment by Fancyflame** [comment]
I have tried running Minecraft server (BDS) on Termux proot using box64.So running box64 in Termux proot is **possible**. But I couldn't compile box64 on Termux without proot, and there still have bugs to run BDS in this environment, which is unexpected segementation fault sometimes will occurred when enabled multithreading(pthread).

**Reply by SayaAEP** [reply]
running bds server on arm64 is heavy stuff. i also want to run bds server on arm64

**Reply by SayaAEP** [reply]
can wine run bds server?

> I have tried running Minecraft server (BDS) on Termux proot using box64.So running box64 in Termux proot is **possible**. But I couldn't compile box64 on Termux without proot, and there still have bugs to run BDS in this environment, which is unexpected segementation fault sometimes will occurred when enabled multithreading(pthread).

**Reply by Fancyflame** [reply]
Hmm, I don't know how to use and where to find wine... XD

**Reply by SayaAEP** [reply]
> Hmm, I don't know how to use and where to find wine... XD

https://github.com/Element-0/ElementZero how about this?

**Reply by SayaAEP** [reply]
> Hmm, I don't know how to use and where to find wine... XD

wine is to run windows application on linux mac android https://www.winehq.org/

**Reply by Fancyflame** [reply]
Thanks, I'll take a look

**Reply by Fancyflame** [reply]
I have seen it, but improving performance is not very important for me and I'm not good at compiling stuff :( . I am more concerned about how to run BDS directly on Android, in other words, how to cross compile box64 to Android. I tried many ways but neither succeeded. Now I can only run bds in proot with performance decrement and potential bugs.

**Reply by SayaAEP** [reply]
> I have seen it, but improving performance is not very important for me and I'm not good at compiling stuff :( . I am more concerned about how to run BDS directly on Android, in other words, how to cross compile box64 to Android. I tried many ways but neither succeeded. Now I can only run bds in proot with performance decrement and potential bugs.

i runned on box64 but stuck locating server 
![image](https://user-images.githubusercontent.com/91467886/154846099-acbeac12-00d1-43f7-bbf8-d3a3440e6c48.png)

**Reply by SayaAEP** [reply]
dpkg --add-architecture i386???

**Reply by SayaAEP** [reply]
https://wiki.debian.org/Multiarch/Implementation

**Reply by SayaAEP** [reply]
> > I have seen it, but improving performance is not very important for me and I'm not good at compiling stuff :( . I am more concerned about how to run BDS directly on Android, in other words, how to cross compile box64 to Android. I tried many ways but neither succeeded. Now I can only run bds in proot with performance decrement and potential bugs.
> 
> i runned on box64 but stuck locating server ![image](https://user-images.githubusercontent.com/91467886/154846099-acbeac12-00d1-43f7-bbf8-d3a3440e6c48.png)

Warning, call to __cxa_thread_atexit_impl(0x2dbde30, 0x7a280017f0, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x3893880, 0x7a40000df8, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x38938c0, 0x7a40000e20, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x3893880, 0x7a54000df8, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x38938c0, 0x7a54000e20, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x3893880, 0x7a60000df8, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x38938c0, 0x7a60000e20, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x3893880, 0x7a4c000df8, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x38938c0, 0x7a4c000e20, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x3893880, 0x7a48000df8, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x38938c0, 0x7a48000e20, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x3893880, 0x7a68000df8, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x38938c0, 0x7a68000e20, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5a860, 0x7a54000d10, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5b2b0, 0x7a54000d38, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5c550, 0x7a54000d60, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2e0a430, 0x7a54000dc0, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5a860, 0x7a60000d10, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5b2b0, 0x7a60000d38, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5c550, 0x7a60000d60, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5a860, 0x7a40000d10, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5b2b0, 0x7a40000d38, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5c550, 0x7a40000d60, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5a860, 0x7a50000d10, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5b2b0, 0x7a50000d38, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5c550, 0x7a50000d60, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x3893880, 0x7a50000df8, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x38938c0, 0x7a50000e20, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5a860, 0x7a68000d10, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5b2b0, 0x7a68000d38, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5c550, 0x7a68000d60, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5a860, 0x7a4c000d10, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5b2b0, 0x7a4c000d38, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2d5c550, 0x7a4c000d60, 0x58baf98) ignored
Warning, call to __cxa_thread_atexit_impl(0x2e0a430, 0x7a68000dc0, 0x58baf98) ignored 


i am still waiting to join the server

**Reply by Fancyflame** [reply]
What's your CPU architecture and environment? mine is arm64 and Termux proot with GNU. I almost always be able to connect to the server successfully except few strange crash. Indeed box64 is still unsteady on Android, if you want better experience, third-party implementation is a good choice, like PocketMine-MP, which I have successfully compiled on Termux without proot and works well.
